src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,191 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private Node<T> currentNode;
+    private Node<T> previousNode;
+    private Node<T> nextNode;
+    private int size;
+
+    public static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.next = next;
+            this.item = item;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> newElement = new Node<T>(null, value, null);
+            head = newElement;
+            tail = newElement;
+        } else {
+            Node<T> newElement = new Node<T>(tail, value, null);
+            Node<T> previousNode = tail;
+            previousNode.next = newElement;
+            tail = newElement;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("The index passed must be greater or equal to 0 "
+                    + "and less or equal to" + size);
+        }
+        if (size - index == 0 || size == 0) {
+            add(value);
+            return;
+        }
+        iterateCurrentNode(index);
+        addNodeInTheMiddle(value);
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        verifyIndex(index);
+        iterateCurrentNode(index);
+        return currentNode.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (size == 0) {
+            throw new IndexOutOfBoundsException("The list is empty");
+        }
+        verifyIndex(index);
+        iterateCurrentNode(index);
+        Node<T> returnItem = new Node<>(currentNode, currentNode.item, currentNode.next);
+        currentNode.item = value;
+        return returnItem.item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        verifyIndex(index);
+        if (index == 0) {
+            return removeFirstNode();
+        }
+        iterateCurrentNode(index);
+        assignPreviousAndNextNode();
+        unlink(currentNode);
+        size--;
+        return currentNode.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        if (size == 0) {
+            throw new NullPointerException("The list is empty");
+        }
+        if (head.item == object || head.item != null && head.item.equals(object)) {
```

Comment:
Be careful with the condition when checking for null. The condition 'head.item == object || head.item != null && head.item.equals(object)' can be simplified to 'object == null ? head.item == null : object.equals(head.item)'.